goreman 0.2.1 (new formula)

kevinburke opened the pull request:
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

kevinburke commented on Formula/goreman.rb:
```diff
@@ -0,0 +1,50 @@
+  url "https://github.com/mattn/goreman/archive/v0.1.1.tar.gz"
+  sha256 "d234ae81a397ab04338f26e30db79811715202413e0a3b9ff20395df7df675df"
+
+  bottle do
```
Apologies but I'm not sure how to initialize these values or even whether I can.

commitay replied:
The bottle block should be removed, it will be added when the formula is merged.

commitay commented on Formula/goreman.rb:
```diff
@@ -0,0 +1,50 @@
+
+  depends_on "go"
+
+  go_resource "github.com/joho/godotenv" do
```
`go_resource` are deprecated and are not acceptable in new formulae, please ask upstream to implement go vendoring.

commitay commented on Formula/goreman.rb:
```diff
@@ -0,0 +1,42 @@
+  url "https://github.com/mattn/goreman/archive/v0.1.1.tar.gz"
+  sha256 "d234ae81a397ab04338f26e30db79811715202413e0a3b9ff20395df7df675df"
+
+  depends_on "go"
```
This is likely a `build` time only dep.

kevinburke replied:
Can you explain what or how I need to modify the code to address this issue?

commitay commented on Formula/goreman.rb:
```diff
@@ -0,0 +1,42 @@
+
+  def install
+    gopath = buildpath/"gopath"
+    ENV["GOPATH"] = gopath
```
```ruby
  def install
    ENV["GOPATH"] = buildpath
```

kevinburke replied:
I get a failure when I do this: 

$GOPATH/go.mod exists but should not

I think Go doesn't like it when you do the symlink like that, where it's alright to add things in a subdirectory.

commitay replied:
```ruby
  def install
    ENV["GOPATH"] = HOMEBREW_CACHE/"go_cache"
    (buildpath/"src/github.com/mattn").install buildpath.children

    cd "src/github.com/mattn" do
      system "go", "build", "-o", bin/"goreman"
      prefix.install_metafiles
    end
  end
```

kevinburke commented:
We've made changes to the goreman source code to vendor all dependencies. I believe I've addressed all of the comments except the ones that cause breaking changes.

Please take another look!

kevinburke commented:
@commitay friendly ping! I think I've addressed the issues you raised :)

kevinburke commented:
Hoping someone can take another look here :) I think I've addressed all of the feedback!

kevinburke commented:
Still waiting for a review from a maintainer, this PR has addressed all
feedback

On Tue, Dec 11, 2018 at 10:09 stale[bot] <notifications@github.com> wrote:

> This issue has been automatically marked as stale because it has not had
> recent activity. It will be closed if no further activity occurs.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Homebrew/homebrew-core/pull/32445#issuecomment-446145455>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAOSI6WGFn14JzL72UTAIQmvlZKUN4x-ks5u34RigaJpZM4W52SH>
> .
>
-- 
Kevin Burke
phone: 925.271.7005 | kev.inburke.com

igas commented:
Hi @kevinburke,

I'll pick up this PR.

To not reiterate line by line I checked it locally, this should be good to go:

```ruby
class Goreman < Formula
  desc "Foreman clone written in Go"
  homepage "https://github.com/mattn/goreman"
  url "https://github.com/mattn/goreman/archive/v0.2.1.tar.gz"
  sha256 "c1ef360fcc92688956bc7a18fae089d78754bd1dde22a89b27228ae5a840cc45"

  depends_on "go" => :build

  def install
    ENV["GOPATH"] = buildpath
    ENV["GO111MODULE"] = "on"

    srcpath = buildpath/"src/github.com/mattn/goreman"
    srcpath.install buildpath.children

    cd srcpath do
      system "go", "build", "-o", bin/"goreman"
      prefix.install_metafiles
    end
  end

  test do
    (testpath/"Procfile").write "web: echo 'hello' > goreman-homebrew-test.out"
    system bin/"goreman", "start"
    assert_predicate testpath/"goreman-homebrew-test.out", :exist?
    assert_match "hello", (testpath/"goreman-homebrew-test.out").read
  end
end
```

Please update PR and I'll merge it.

kevinburke commented:
Done, thank you!

igas commented:
Thank you @kevinburke!
